modules/unsupported/process-feature/src/main/java/org/geotools/process/vector/FilteringVectorProcess.java

```diff
@@ -0,0 +1,423 @@
+                Operations aggregation) {
+            super(iterator);
+            this.groupByAttributes = groupByAttributes;
+            this.operationAttribute = operationValue;
+            this.aggregation = aggregation;
+        }
+
+        @Override
+        public boolean hasNext() {
+            List<Object> groupingValues = new ArrayList<>(groupByAttributes.size());
+            Map<Object, Feature> beingFiltered = new HashMap<>();
+            List<Double> valuesToCompare = new ArrayList<>();
+            while (super.hasNext()) {
+                Feature f = super.next();
+                if (beingFiltered.size() == 0) {
+                    // no features in the list this is the first of the group
+                    // takes the values to check the following features if belong to the same group
+                    // features with null values are skipped
+                    if (addGroupingValues(groupingValues, f))
+                        addDoubleValueFromFeature(f, valuesToCompare, beingFiltered);
+                } else {
+                    // is the feature in the group?
+                    if (featureComparison(groupingValues, f)) {
+                        addDoubleValueFromFeature(f, valuesToCompare, beingFiltered);
+                    } else {
+                        ((PushBackFeatureIterator) delegate).pushBack();
+                        break;
+                    }
+                }
+            }
+            next = doFiltering(beingFiltered, valuesToCompare);
+            return next != null;
+        }
+
+        private Feature doFiltering(
+                Map<Object, Feature> beingFiltered, List<Double> beingEvaluated) {
+            Object key;
+            // searches the min or max inside the values list and retrieve the feature
+            // to be returned from the Map.
+            if (beingFiltered.size() > 0 && beingEvaluated.size() > 0) {
+                if (this.aggregation.equals(Operations.MIN)) key = computeMin(beingEvaluated);
+                else key = computeMax(beingEvaluated);
+
+                return beingFiltered.get(key);
+            } else {
+                return null;
+            }
+        }
+
+        private Number computeMin(List<Double> beingEvaluated) {
+            return Collections.min(beingEvaluated);
+        }
+
+        private Number computeMax(List<Double> beingEvaluated) {
+            return Collections.max(beingEvaluated);
+        }
+
+        private boolean featureComparison(List<Object> groupingValues, Feature f) {
+            List<Object> toCompareValues = new ArrayList<>(groupingValues.size());
+            for (PropertyName p : groupByAttributes) {
+                toCompareValues.add(p.evaluate(f));
+            }
+            if (groupingValues.size() > 0 && groupingValues.equals(toCompareValues)) return true;
+            return false;
+        }
+
+        private boolean addGroupingValues(List<Object> groupingValues, Feature f) {
```

Comment:
This is not really adding the values to the list, it's more like gathering them. Seems also dangerous, I see no cleaning happening, if a feature is skipped for any reason, the groupingValues would keep on accumulating?
IMHO better to have this method return a feature list of values, or null if they are not found.

Reply:
fixed